### ISSUE TITLE
Frontend integration tests use port 8000 when it's not available

### ISSUE THREAD

**Issue by itbabu** [divio/django-cms#5424]
I was trying to run the frontend integration tests for  #5391

Before I was working on an other project and I forgot a ```runserver``` on port 8000
and then I run the integration tests.
The test server was up and it was using an other port (my guess), but the tests were still using 8000 and so they failed.

Here my shell output:

```shell
➜  django-cms git:(develop) gulp tests:integration --clean                 
```

```console
[11:49:27] Using gulpfile ~/workspace/penta/django-cms/gulpfile.js
[11:49:27] Starting 'tests:integration'...
[11:49:27] Starting a server

(migrations, etc) 

A admin user (username: admin, password: admin)has been created.

[11:50:57] CasperJS: Test file: /home/marco/workspace/penta/django-cms/cms/tests/frontend/integration/setup.js
[11:50:57] CasperJS: Current viewport size is 1280x1024.
[11:50:57] CasperJS: Test file: /home/marco/workspace/penta/django-cms/cms/tests/frontend/integration/loginAdmin.js
[11:50:57] CasperJS: # User Login (via Admin Panel)
[11:50:57] CasperJS: PASS User Login (via Admin Panel) (NaN test)
[11:50:58] CasperJS: Resource failed to load: http://localhost:8000/en/admin/login/
[11:50:58] CasperJS: 404 page found: http://localhost:8000/en/admin/login/
[11:50:58] CasperJS: The currently set cookies are: [{"domain":"localhost","expires":"lun, 27 giu 2016 09:50:58 GMT","expiry":1467021058,"httponly":false,"name":"djdt","path":"/","secure":false,"value":"show"}]
[11:50:58] CasperJS: FAIL The CMS is available and admin panel title is correct
[11:50:58] CasperJS: #    type: assertTitle
#    file: /home/marco/workspace/penta/django-cms/cms/tests/frontend/integration/loginAdmin.js
[11:50:58] CasperJS: #    subject: "Page not found at /en/admin/login/"
#    pattern: "/Log in | Django site admin/g"
[11:50:58] CasperJS: assert failed: The CMS is available and admin panel title is correct
[11:50:58] Failure
[11:50:58] 'tests:integration' errored after 1.52 min
[11:50:58] Error: 1
    at formatError (/usr/local/lib/node_modules/gulp-cli/lib/versioned/^3.7.0/formatError.js:20:10)
    at Gulp.<anonymous> (/usr/local/lib/node_modules/gulp-cli/lib/versioned/^3.7.0/log/events.js:41:15)
    at Gulp.emit (events.js:107:17)
    at Gulp.Orchestrator._emitTaskDone (/home/marco/workspace/penta/django-cms/node_modules/gulp/node_modules/orchestrator/index.js:264:8)
    at /home/marco/workspace/penta/django-cms/node_modules/gulp/node_modules/orchestrator/index.js:275:23
    at finish (/home/marco/workspace/penta/django-cms/node_modules/gulp/node_modules/orchestrator/lib/runTask.js:21:8)
    at cb (/home/marco/workspace/penta/django-cms/node_modules/gulp/node_modules/orchestrator/lib/runTask.js:29:3)
    at /home/marco/workspace/penta/django-cms/node_modules/djangocms-casper-helpers/gulp/index.js:262:17
    at tryCatcher (/home/marco/workspace/penta/django-cms/node_modules/djangocms-casper-helpers/node_modules/bluebird/js/release/util.js:16:23)
    at Promise._settlePromiseFromHandler (/home/marco/workspace/penta/django-cms/node_modules/djangocms-casper-helpers/node_modules/bluebird/js/release/promise.js:502:31)
    at Promise._settlePromise (/home/marco/workspace/penta/django-cms/node_modules/djangocms-casper-helpers/node_modules/bluebird/js/release/promise.js:559:18)
    at Promise._settlePromise0 (/home/marco/workspace/penta/django-cms/node_modules/djangocms-casper-helpers/node_modules/bluebird/js/release/promise.js:604:10)
    at Promise._settlePromises (/home/marco/workspace/penta/django-cms/node_modules/djangocms-casper-helpers/node_modules/bluebird/js/release/promise.js:679:18)
    at Async._drainQueue (/home/marco/workspace/penta/django-cms/node_modules/djangocms-casper-helpers/node_modules/bluebird/js/release/async.js:138:16)
    at Async._drainQueues (/home/marco/workspace/penta/django-cms/node_modules/djangocms-casper-helpers/node_modules/bluebird/js/release/async.js:148:10)
    at Immediate.Async.drainQueues [as _onImmediate] (/home/marco/workspace/penta/django-cms/node_modules/djangocms-casper-helpers/node_modules/bluebird/js/release/async.js:17:14)
    at processImmediate [as _immediateCallback] (timers.js:367:17)

**Comment by vxsx** [comment]
i don't think i will be able to easily detect other servers running on the same port, but as a fix we could use less common port by default (e.g. 6789 or smth) and provide a possibility to specify it via `--port` 

do you think that would be sufficient @itbabu ?

**Comment by itbabu** [comment]
Maybe a verbose message when starting the server?
Like ```Starting a server on port 6789``` or ```Port 6789 already in use. Starting the server on port 6797``
The port change was not obvious to me while looking at the tests logs

**Comment by vxsx** [comment]
Should be fixed in `release/3.3.x`
